paddle/pybind/pybind.cc

```diff
@@ -25,9 +26,143 @@ namespace pd = paddle::framework;
 
 USE_OP(add_two);
 
+struct PlaceDebugString : public boost::static_visitor<std::string> {
+  std::string operator()(const paddle::platform::GPUPlace& place) const {
+    return "GPU(" + std::to_string(place.device) + ")";
+  }
+
+  std::string operator()(const paddle::platform::CPUPlace& place) const {
+    return "CPU";
+  }
+};
+
+template <typename T>
```

Comment:
Other I will follow comments.

1. Compile `CastToPyBufferImpl` and `TensorToPyBuffer` together.
2. Move them into a header file.
3. use `friend` to get `Place` of tensor.